### PR TITLE
Auth next updates

### PR DESCRIPTION
**Pull request by paulbellamy**
Fixes https://github.com/stellar/soroban-tools/issues/351

Depends on:
- [x] https://github.com/stellar/js-soroban-client/pull/47
- [x] https://github.com/esteblock/soroban-react/pull/21
- [ ] https://github.com/stellar/quickstart/pull/424

# TODO

- [x] update rust deps
- [x] update soroban_token_spec.wasm from soroban-examples
- [x] check it all builds
- [x] fix rust tests
	- [x] https://github.com/stellar/rs-soroban-sdk/pull/856
- [x] stop doing approve & transfer workflow
	- [x] update the contract
	- [x] update the web app
	- [x] figure out how to build addresses in the web app
		- [x] add a helper for this to js-soroban-client
- [x] Check that `sendTransaction` is doing the prepareTransaction, auth and footprint updates
- [x] update soroban-client dep
- [x] update soroban-react deps
- [x] update quickstart image hash
- [ ] Test it all works

### CODE REVIEW

File: contracts/crowdfund/src/lib.rs [hunk 1]
```diff
@@ -180,7 +177,9 @@ impl Crowdfund {
         get_user_deposited(&e, &user)
     }
 
-    pub fn deposit(e: Env, user: Identifier, amount: i128) {
+    pub fn deposit(e: Env, user: Address, amount: i128) {
+        // We don't need to check the user signature here, because the token.xfer below will check
```
Comment (dmkozh) [inline contracts/crowdfund/src/lib.rs:181]: This is not correct - the context-less `xfer` can be frontrun and hence the xfer will happen without a recording. Hence `user.require_auth()` needs to be called here.

File: contracts/crowdfund/src/test.rs [hunk 2]
```diff
@@ -3,10 +3,9 @@
...
 use rand::{thread_rng, RngCore};
-use soroban_auth::{Identifier, Signature};
 use soroban_sdk::{
-    testutils::{Accounts, Ledger},
-    xdr, AccountId, BytesN, Env,
+    testutils::{Address as AddressTestTrait, Ledger},
+    Address, BytesN, Env,
 };
 
 fn generate_contract_id() -> [u8; 32] {
```
Comment (dmkozh) [inline contracts/crowdfund/src/test.rs:11]: nit/non-blocking: you shouldn't really need this - just pass `None` to the cotnract registration fn and use the returned value.

File: contracts/crowdfund/src/lib.rs (hunk 1 above)
Comment (paulbellamy) [reply contracts/crowdfund/src/lib.rs:181]: Ok.. I think there's something about auth-next I don't understand, then.

Comment (paulbellamy) [reply contracts/crowdfund/src/lib.rs]: Discussed elsewhere.

File: README.md [hunk 3]
```diff
@@ -64,9 +64,9 @@ npm run dev
 
 Open [http://localhost:3000](http://localhost:3000) with your browser to see the result.
 
-Note: Before you can "Approve transfer & Back this project", you'll need to have
-some EXT (example token) in your freighter wallet. There is a "Mint 100 EXT"
-button, which will gift you 100 EXT tokens for that purpose.
+Note: Before you can "Back this project", you'll need to have some EXT (example
+token) in your freighter wallet. There is a "Mint 100 EXT" button, which will
+gift you 100 EXT tokens for that purpose.
```
Comment (Shaptic) [inline README.md:70]: The changes seem fine to me, but it'd be great to have the "what/why" description of this change, as it's a little hard for me to follow the changes in the context of auth-next and think through PR approval.

File: initialize.sh [hunk 4]
```diff
@@ -5,65 +5,83 @@ set -e
...
+  echo Add the $NETWORK network to cli client
+  soroban config network add "$NETWORK" \
+    --rpc-url "$SOROBAN_RPC_URL" \
+    --network-passphrase "$SOROBAN_NETWORK_PASSPHRASE"
+fi
+
+if [ ! -f ".soroban/identities/token-admin.toml" ]; then
+  echo Create the token-admin identity
+  mkdir -p ".soroban/identities"
+  echo "secret_key = \"$TOKEN_ADMIN_SECRET\"" > ".soroban/identities/token-admin.toml"
```
[the remark this answers is not in the document]
Comment (paulbellamy) [reply initialize.sh:44]: We could. `soroban config identity address ...` did not exist when I wrote this. We would also need to inject it into the js.

File: initialize.sh [hunk 5]
```diff
@@ -5,65 +5,83 @@ set -e
...
     --wasm target/wasm32-unknown-unknown/release/soroban_crowdfund_contract.wasm
 )"
 echo "$CROWDFUND_ID" > .soroban/crowdfund_id
 
 echo "Contract deployed succesfully with ID: $CROWDFUND_ID"
 
+echo "Wait 5s for a ledger to close"
+# TODO: Figure out why we need this. Without it, soroban-rpc
+# simulateTransaction says AccessToUnknownLedgerEntry...
+sleep 5
```
[the remark this answers is not in the document]
Comment (paulbellamy) [reply initialize.sh:73]: I think it was the ingestion bug tamir fixed. So we probably don't need this anymore.

File: README.md (hunk 3 above)
Comment (paulbellamy) [reply README.md:70]: So, almost all of the changes are just updating the sdk usage and cli usage for the new stuff. The auth related stuff are:
- Replacing Identifier types with Address
- [The line where we call `require_auth`](https://github.com/stellar/soroban-example-dapp/pull/87/files#diff-b9327a7b989f7d347d6b1e702583634f9eedf42542de3f1220e6ec2bf9de66acR181), is what triggers the auth when we preflight the txn.

Generally, this app doesn't use auth much. The only thing we used it for previously was the approve & transfer. With Auth-next we don't need the approve anymore, we can do a delegated-xfer. Which is what happens in the `deposit` method. It's somewhat magic.

Comment (Shaptic) [reply README.md:70]: perfect, got it, thanks for explaining!

**Review by Shaptic — APPROVED** [review]
:shipit:

**Pull request merged**